### ISSUE TITLE
direnv recursively executes in chroot/pivot_root

### ISSUE THREAD

**Issue by eadwu** [direnv/direnv#817]
**Describe the bug**
In the case of switching directories to `pivot_root` with a followup `chdir` or `chroot`, typically the final directory would end up being `/`, when `cd`ing back to the `direnv` directory, the `direnv` hook runs again.

**To Reproduce**
Inside `.envrc`

```bash
layout_imperial() {
  local left="${1}"
  [[ $# -gt 0 ]] && shift

  local right="${1}"
  [[ $# -gt 0 ]] && shift

  if [ ! -d "${left}" ] || [ ! -d "${right}" ];
  then
    log_error "Required directories do not exist"
    return 1
  fi

  local UNIDIS="$(command -v unidis)"
  if [ -z "${UNIDIS}" ];
  then
    log_error "Executable `unidis` not found"
  fi

  "${UNIDIS}" --remount-tmp --right "${right}" --unionfs fuseunionfs --update-path \
    "${left}" "${SHELL}"
}

layout_imperial "/" ".arch/root"
```

**Expected behavior**
Only a single `direnv` run, regardless of levels of nested views, or maybe this should be kept but the ability to prevent nested execution.

**Environment**
 - OS: NixOS
 - Shell: zsh
 - Direnv: 2.28.0

**Additional context**
`unidis` ugly source code is located here, https://github.com/eadwu/imperial/tree/master/lib

In a nutshell, `unidis` just merges two directory paths to create a FHS-compatible root view and uses `pivot_root` and `chroot` syscalls.

You can assume `.arch/root` is a fully setup Arch Linux root folder only missing the boot partition (which would be mounted at `.arch/root/boot`. For the purposes of the recursive execution, it doesn't really matter, `"/" "/"` would work just fine to reproduce anyway.

**Comment by chenlijun99** [comment]
Would also be interested to find a workaround to this issue.

**Comment by chenlijun99** [comment]
#1048 has some related discussions.
